extensions/adobe/experience/customerJourneyManagement/messageexecution.schema.json

```diff
@@ -0,0 +1,58 @@
+{
+  "meta:license": [
+    "Copyright 2020 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/experience/customerJourneyManagement/messageexecution",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Adobe CJM ExperienceEvent - MessageExecution Details",
+  "description": "MessageExecution details for the Adobe CJM ExperienceEvent.",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/experienceevent"],
+
+  "definitions": {
+    "messageExecution": {
+      "properties": {
+        "@id": {
+          "title": "MessageExecution ID",
+          "type": "string",
+          "description": "The CJM activity originating this message."
+        },
+        "xdm:messageID": {
+          "title": "Message ID",
+          "type": "string",
+          "description": "The parent Message ID of a MessageExecution originating this message."
+        },
+        "xdm:messageLabel": {
+          "title": "MessageExecution Label",
+          "type": "string",
+          "description": "A human-friendly name of the Message ID."
+        },
+        "xdm:externalApplicationID": {
+          "title": "External Application ID",
+          "type": "string",
+          "description": "The ID of the external Application entity triggering the MessageExecution."
+        },
+        "xdm:externalApplication": {
```

Comment:
Should we have dedicated field like JourneyId for referring to the external application which initiated the message execution. Or keep it generic with field 'xdm:externalApplicationID'.

Reply:
Could the ID be namespaced?
IIRC XDM had a datasource or similar that had notion of NS+ID

Reply:
the combination of externalAppicationID and externalApplication provides a namespaced ID

Reply:
and I prefer the generic approach rather than a journey specific one